viking/src/tools/check.rs

```diff
@@ -144,34 +154,44 @@ fn check_all(
                 decomp_symtab,
                 function,
             )?;
-            if !ok {
+            if matches!(ok, CheckResult::MismatchError) {
                 failed.store(true, std::sync::atomic::Ordering::Relaxed);
             }
+            else if matches!(ok, CheckResult::MatchWarn) {
+                matching_functions.lock().unwrap().push(function.addr);
```

Comment:
This severely hurts parallelism and forces people who do *not* want to use the new --update-matching flag to eat the cost, which is unacceptable. Please add a check for `update_matching`.

Reply:
> This severely hurts parallelism ...
Is there any better way to collect the results from all of the threads at once?

Side note: In a "clean" decompile project, this path should never trigger. It only gets executed when the function matches, but it's specified as non-matching in the database.

Reply:
> Is there any better way to collect the results from all of the threads at once?

There are concurrent data structures you could use but honestly a mutex is probably fine. However, what is *not* fine is paying for this feature even when it's not being used (doesn't matter if BotW isn't affected by this in practice, that's a general principle when you design optional features).